src/thunderbots/software/ai/hl/stp/play/play.h

```diff
@@ -136,33 +136,36 @@ class Play
 };
 
 /**
- * This macro is used by Plays that are derived from the Abstract Play class above. Its
- * purpose is to create a Factory for the implemented Play, and declare a static variable
- * for the factory so that the Play is automatically registered in the Play
- * registry.
+ * This templated play factory class is used by Plays that are derived from the Abstract
+ * Play class above. Its purpose is to create a Factory for the implemented Play and
+ * automatically register the play in the play registry.
  *
  * Declaring the static variable will also cause it to be initialized at the start of the
  * program (because it's static). This will immediately call the constructor, which adds
  * a pointer to the Factory to the Play registry. From then on, the rest of the program
  * can use the registry to find all the Plays that are available (and register with this
- * macro).
+ * templated class).
  *
- * @param play_class_name The classname of the Play to be added to the registry. For
- * example, to add new class called MovePlay that inherits from Play, the following line
- * should be added to the end of the .cpp file (without the quotations):
- * "REGISTER_PLAY(MovePlay)"
+ * @param <T> The class of the Play to be addred to the registry. For example, to add a
+ * new class called MovePlay that inherits from Play, the following line should be added
+ * to the end of the .cpp file (without the quotations):
```

Comment:
nit; but looks like templated params should be `@tparam` https://stackoverflow.com/questions/13359217/how-to-document-c-templates-and-template-metafunctions-with-doxygen